frontend/src/component/Gallery.tsx

```diff
@@ -2,11 +2,10 @@ import {Workout} from "../model/Workout";
 import WorkoutCard from "./WorkoutCard";
 import "./Gallery.css"
 
-
 type GalleryProps = {
 
     workouts: Workout[]
-    deleteWorkout: (workout: Workout) => void
+    deleteWorkout: (id: string) => void
```

Comment:
deleteWorkout could return a Promise, then you can wait for it in this component